runtime/gc_base/StringTable.cpp

```diff
@@ -554,7 +554,19 @@ j9gc_createJavaLangString(J9VMThread *vmThread, U_8 *data, UDATA length, UDATA s
 	j9object_t charArray;
 	UDATA allocateFlags = J9_GC_ALLOCATE_OBJECT_NON_INSTRUMENTABLE;
 	UDATA unicodeLength;
-	bool isCompressable = false;
+
+	bool isCompressible = true;
+
+	for (UDATA i = 0; i < length; ++i) {
+		if (data[i] > 127) {
+			isCompressible = false;
+			break;
+		}
+	}
+
+	if (isCompressible) {
+		stringFlags |= J9_STR_ASCII;
+	}
```

Comment:
Performance of this API can further be improved by by not having to execute this piece of code within this API. Profiling a large webserver workload tells us that this API gets exercised quite a but during exception backtrace generation. In such a context the strings that typically get passed through to this API are strings representing the full class path, class name, and file name. These are `U_8*` strings obtained from data structures such as `J9ROMClass`, etc. which get passed as arguments to this API.

We should be able to execute this loop during startup at a minimal cost and add the `J9_STR_ASCII` flag once, rather than repeating the calculation here every time. This is a startup vs. throughput tradeoff which will have to be measured. If the project leads agree we can attempt to prototype this in another PR and look at the trade offs. @DanHeidinga @vijaysun-omr FYI for your thoughts. I can open another issue to track this work.

Reply:
Note that having these `U_8*` UTF8 strings tagged as ASCII once at startup has other benefits in various other APIs related to `-XX+CompactStrings` which we can take advantage of. This is just one example of such an API.

Reply:
>  I can open another issue to track this work.

+1.  Let's get an issue opened for that design as the primary place this matters is for method and classnames.

Reply:
I've opened up #7766 to track this work.